rust-engine/src/lib.rs

```diff
@@ -11,8 +11,8 @@
 pub mod ast;
 pub mod backends;
 pub mod hax_io;
-pub mod lean;
 pub mod names;
 pub mod ocaml_engine;
+pub mod phase;
```

Comment:
Do all of these need to be public?

Reply:
Generally, the visibility of items was done so that backends could be separate crates at some point.
This generally would be good IMO; that's what we used to have in OCaml, one package for each crate.

So here, `phase` should be public, otherwise, a backend won't be able to call a phase.